src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -76,9 +76,20 @@ protected function verboseLog($message)
      */
     protected function createPdoConnection($dsn, $username = null, $password = null, array $options = [])
     {
+        $adapterOptions = $this->getOptions();
+
+        if (!isset($adapterOptions['attr_errmode'])) {
+            $adapterOptions['attr_errmode'] = PDO::ERRMODE_EXCEPTION;
+        }
```

Comment:
```suggestion
        $adapterOptions = $this->getOptions() + [
            'attr_errormode' => PDO::ERRMODE_EXCEPTION;
        ];
```

Reply:
There doesn't seem to be reason to have this up here. Move it down to where the attributes are used.

Reply:
It was put above to minimize the amount of code under the try block.

For your suggestion, just to confirm you want phinx to mandate that `attr_errmode` MUST be set to `\PDO::ERRMODE_EXCEPTION`, even if the user sets it to something else for some reason or other.

Reply:
No, this is merging the arrays.

Reply:
@othercorey The merge has the wrong order, otherwise looks good.

Reply:
May I ask about the preference over array merging for here instead of using `isset`? Is it just personal preference on the part of cakephp folks, or a technical reason for it as well? Just trying to get a handle for future PRs and avoiding repeated review comments.

Reply:
It seems the right order. You want it set only as the default.

We use merging for options defaults when possible because it is easier to add new ones and store the defaults in a property.

Reply:
Ah ok.

@MasterOdin this is the right operator for assoc array merging, especially for performance and reliability.

Reply:
@othercorey fair enough. I will keep that in mind.

Anyway, suggestion committed.